src/block_cache/block_tier.c

```diff
@@ -97,43 +97,53 @@ __wt_blkcache_tiered_open(
 
 /*
  * __blkcache_find_open_handle --
- *     If the block manager's handle array already has an entry for the given object, return it.
+ *     If the block manager's handle array already has an entry for the given object, return it. If
+ *     caller is going to read from the handle, increment the read count while we have the handle
+ *     table locked.
  */
 static void
-__blkcache_find_open_handle(WT_BM *bm, uint32_t objectid, WT_BLOCK **blockp)
+__blkcache_find_open_handle(WT_BM *bm, uint32_t objectid, bool reading, WT_BLOCK **blockp)
 {
     u_int i;
 
+    /* Must be called with bm->handle_array_lock locked */
```

Comment:
Can you use Peter's new assertion/ownership changes?

Reply:
`handle_array_lock` is a RW lock; the assertions that I added are just for spinlocks. But that being said, it would be nice if the comment could explicitly specify whether "locked" means a read lock or a write lock.

Reply:
Good point. I've updated the comment.